src/components/Modal/Modal.tsx

```diff
@@ -86,19 +123,19 @@ const Modal: FC<ModalProps> = ({
             Expired Date
             <Input
               type="datetime-local"
-              value={expiredDate}
-              min={getTenMinAfter(createdDate)}
+              value={modalData.expiredDate}
+              min={getTenMinAfter(modalData.createdDate)}
               max={getTenYearsAfter()}
               onChange={handleExpiredDateChange}
             />
           </label>
           <div className={styles.buttons}>
-            <Button text="Cancel" style="red" onClick={closeModal} />
             <Button
-              text="Save"
-              type="submit"
-              disabled={!createdDate || !expiredDate}
+              text="Cancel"
+              style="red"
+              onClick={() => setShowModal(false)}
```

Comment:
Remove logic from JSX

Reply:
Fixed - in parrent component create 2 functions: openModal and closeModal. And send closeModal throught props to Modal component instead of setState function.